joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 125

robot_controller:
  type: position_controllers/JointTrajectoryController

  joints: 
  - joint_1
  - joint_2
  - joint_3
  - joint_4
  - joint_5
  - joint_6
  - joint_7

  # gains:
  #   joint_1: {p: 1000, d: 30, i: 1, i_clamp: 0}
  #   joint_2: {p: 1000, d: 30, i: 1, i_clamp: 0}
  #   joint_3: {p: 1000, d: 30, i: 1, i_clamp: 0}
  #   joint_4: {p: 1000, d: 30, i: 1, i_clamp: 0}
  #   joint_5: {p: 1000, d: 30, i: 1, i_clamp: 0}
  #   joint_6: {p: 1000, d: 30, i: 1, i_clamp: 0}
  #   joint_7: {p: 1000, d: 30, i: 1, i_clamp: 0}

  constraints:
    goal_time: 0.6
    stopped_velocity_tolerance: 0.05
    joint_1: {trajectory: 0.2, goal: 0.1}
    joint_2: {trajectory: 0.2, goal: 0.1}
    joint_3: {trajectory: 0.2, goal: 0.1}
    joint_4: {trajectory: 0.2, goal: 0.1}
    joint_5: {trajectory: 0.2, goal: 0.1}
    joint_6: {trajectory: 0.2, goal: 0.1}
    joint_7: {trajectory: 0.2, goal: 0.1}

  stop_trajectory_duration: 0.5
  state_publish_rate: 125
  action_monitor_rate: 20

gripper_controller:
  # type: effort_controllers/JointGroupEffortController
  # type: position_controllers/JointGroupPosition Controller
  type: effort_controllers/JointTrajectoryController

  joints: 
  - gripper_joint_l
  - gripper_joint_r
  
  constraints:
    goal_time: 0.6
    stopped_velocity_tolerance: 0.05
    gripper_joint_l: {trajectory: 0.2, goal: 0.1}
    gripper_joint_r: {trajectory: 0.2, goal: 0.1}

  gains:
    gripper_joint_l: {p: 100, d: 0, i: 1, i_clamp: 20}
    gripper_joint_r: {p: 100, d: 0, i: 1, i_clamp: 20}
